pkg/controller/identity/iamuser/controller.go

```diff
@@ -0,0 +1,251 @@
+	}
+
+	observed, err := e.client.GetUserRequest(&awsiam.GetUserInput{
+		UserName: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errGet)
+	}
+
+	if observed.User == nil {
+		return managed.ExternalObservation{}, errors.New(errSDK)
+	}
+
+	user := *observed.User
+	current := cr.Spec.ForProvider.DeepCopy()
+	iam.LateInitializeUser(&cr.Spec.ForProvider, &user)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = v1alpha1.IAMUserObservation{
+		Arn:    aws.StringValue(user.Arn),
+		UserID: aws.StringValue(user.UserId),
+	}
+
+	groupsResp, err := e.client.ListGroupsForUserRequest(&awsiam.ListGroupsForUserInput{
+		UserName: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGet)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists: true,
+		ResourceUpToDate: iam.CompareGroups(cr.Spec.ForProvider, groupsResp.ListGroupsForUserOutput.Groups) &&
+			aws.StringValue(cr.Spec.ForProvider.Path) == aws.StringValue(user.Path),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMUser)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	_, err := e.client.CreateUserRequest(&awsiam.CreateUserInput{
+		Path:                cr.Spec.ForProvider.Path,
+		PermissionsBoundary: cr.Spec.ForProvider.PermissionsBoundary,
+		Tags:                iam.BuildIAMTags(cr.Spec.ForProvider.Tags),
+		UserName:            aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.IAMUser)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	groupsResp, err := e.client.ListGroupsForUserRequest(&awsiam.ListGroupsForUserInput{
+		UserName: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+
+	if !iam.CompareGroups(cr.Spec.ForProvider, groupsResp.Groups) {
+
+		// Remove user from current groups
+		for _, val := range groupsResp.Groups {
+			if _, err = e.client.RemoveUserFromGroupRequest(&awsiam.RemoveUserFromGroupInput{
+				GroupName: val.GroupName,
+				UserName:  aws.String(meta.GetExternalName(cr)),
+			}).Send(ctx); err != nil {
+				return managed.ExternalUpdate{}, errors.Wrap(err, errRemoveFromGroup)
+			}
+		}
```

Comment:
Does this mean that there will be a brief window in which this user is removed from all of the groups we want them to be in, before we re-add them immediately below? It seems like it would be safer to calculate the diff and only remove them from the groups that we know we no longer want them in.

Reply:
@sahil-lakhwani I think this needs to be addressed before we merge. I'd be happy with either only removing the groups that the user should actually be removed from, or just removing group membership from this CR altogether.

Reply:
@negz I just wanted to note here that I was not able to track down how this looked previously, so not sure if comment was addressed sufficiently or not